sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/MLflowDataObject.scala

```diff
@@ -0,0 +1,100 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.smartdatalake.workflow.dataobject
+
+import io.smartdatalake.config.SdlConfigObject.DataObjectId
+import io.smartdatalake.config.{ FromConfigFactory, InstanceRegistry }
+import io.smartdatalake.util.misc.SmartDataLakeLogger
+import com.typesafe.config.Config
+import io.smartdatalake.util.hdfs.PartitionValues
+import io.smartdatalake.util.mlflow.{ MLflowPythonSparkEntryPoint, MLflowPythonUtil }
+import io.smartdatalake.workflow.ActionPipelineContext
+import org.apache.spark.sql.{ DataFrame }
+
+// TODO: model according to experiment definition
+case class MLflowExperiment(experimentName: String)
+
+/**
+ * [[DataObject]] for interaction with MLflow
+ *
+ * @param trackingURI Uri of the MLflow server. Default is http://localhost:5000.
+ * @param experimentName The name of the experiment stored in MLflow.
+ */
+case class MLflowDataObject(
+                             override val id: DataObjectId,
+                             trackingURI: String,
+                             experimentName: String,
+                             modelName: Option[String] = None,
+                             modelDescription: Option[String] = None,
+                             override val metadata: Option[DataObjectMetadata] = None
+                           )(@transient implicit val instanceRegistry: InstanceRegistry)
+  extends DataObject
+    with CanCreateSparkDataFrame
+    with SmartDataLakeLogger {
+
+  // entry point for accessing dynamically Java objects living inside the JVM
+  var entryPoint: Option[MLflowPythonSparkEntryPoint] = None
+  var pythonMLflowApi: Option[MLflowPythonUtil] = None
+
+  //
+  var experimentId: Option[String] = None
+
+  override def prepare(implicit context: ActionPipelineContext): Unit = {
+    super.prepare
+    // create entry point
+    entryPoint = Some(new MLflowPythonSparkEntryPoint(context.sparkSession, options))
+    if (entryPoint.isEmpty) {
+      throw MLflowException("Creation of MLflowPythonSparkEntryPoint was not successful")
+    } else {
+      logger.info("Created MLflowPythonSparkEntryPoint")
+      pythonMLflowApi = Some(MLflowPythonUtil(entryPoint.get, trackingURI))
+      experimentId = pythonMLflowApi
+        .getOrElse(throw MLflowException("PythonUtil for MLflow not ready"))
+        .getOrCreateExperimentID(experimentName)
+      logger.info(s"Working with MLflow experiment $experimentName with ID $experimentId")
+    }
+
+  }
+
+  override def getSparkDataFrame(
+                                  partitionValues: Seq[PartitionValues] = Seq()
+                                )(implicit context: ActionPipelineContext): DataFrame = {
+    implicit val session = context.sparkSession
+    import session.implicits._
+
+    // TODO: fetch latest run information and return as DataFrame
+    val df: DataFrame = Seq.empty[MLflowExperiment].toDF()

```

Comment:
If we return run information here, can we use this input DataFrame in MLFlowPredictAction to choose the run version to be used for prediction? We could then have some Spark expression configured for MLFlowPredictAction to choose one run from the DataFrame...

Reply:
This contradicts a bit the previous comment. Run information is written to the 'runInfo' DataObject. So yes, this could be used for input in the `MLFlowPredictAction`.